src/decoder_fmmidi.cpp

```diff
@@ -50,15 +49,11 @@ int read_func(void* instance) {
 	return fmmidi->file_buffer[fmmidi->file_buffer_pos++];
 }
 
-bool FmMidiDecoder::Open(FILE* file) {
-	this->file = file;
-
+bool FmMidiDecoder::Open(Filesystem::InputStream stream) {
```

Comment:
If the decoder doesn't take ownership, it should pass by reference to avoid a refcount.

`unique_ptr` makes this easier as you'll get compiler errors unless you either pass it by reference for `std::move` it.